src/coffee/io/writer.coffee

```diff
@@ -0,0 +1,105 @@
+_ = require 'underscore'
+Csv = require 'csv'
+path = require 'path'
+Promise = require 'bluebird'
+iconv = require 'iconv-lite'
+fs = Promise.promisifyAll require('fs')
+Excel = require 'exceljs'
+
+DEBUG = false
+debugLog = if DEBUG then console.log else _.noop
+
+class Writer
+
+  constructor: (@options = {}) ->
+    debugLog "WRITER::options:", JSON.stringify(@options)
+    @options.defaultEncoding = "utf8"
+
+    # write to file or to stdout?
+    if @options.outputFile
+      debugLog "WRITER::stream file %s", @options.outputFile
+      @outputStream = fs.createWriteStream @options.outputFile
+    else
+      debugLog "WRITER::stream stdout"
+      @outputStream = process.stdout
+
+    # if we use xlsx export - create workbook first
+    if @options.format == 'xlsx'
+      @options.workbookOpts = {
+        stream: @outputStream,
+        useStyles: true,
+        useSharedStrings: true
+      }
+
+      @workbook = new Excel.stream.xlsx.WorkbookWriter @options.workbookOpts
+      @worksheet = @workbook.addWorksheet('Products')
+
+  encode: (string) =>
+    if @options.encoding == @options.defaultEncoding
+      return string
+
+    if not iconv.encodingExists @options.encoding
+      throw new Error 'Encoding does not exist: '+ @options.encoding
+
+    iconv.encode string, @options.encoding
+
+  # create header
+  setHeader: (header) =>
+    debugLog("WRITER::writing header %s", header)
+
+    if @options.format == 'xlsx'
+      @_writeXlsxHeader header
+    else
```

Comment:
This is related to your first comment about the format, I would say support exporting to one format only with .eg. `--format csv`. This will also prevent the need to check options mid-way in the process.